### PR TITLE
CMR-8089: Updates background job to not use hard coded mappings

### PR DESCRIPTION
**Pull request by abbottry**
Queries to OpenSearch, including the background jobs to process holdings no longer use hard coded mappings -- it instead expects proper and updated metadata to provide the necessary information to process each request.

### CODE REVIEW

File: app/controllers/granules_controller.rb [hunk 1]
```diff
@@ -1,131 +1,131 @@
...
-
-            @number_of_hits = Granule.hits granules
-            @client_id = params[:clientId] || 'unknown'
-
-            if request.format == :html
-              @client_id = 'our_html_ui'
-              @granule_models = Granule.to_html_model granules
-            end
+    @time = Benchmark.realtime do
+      # do not allow the unrestricted granule seach if via the granule search form unless accessed as step 2 of a 2 step
```
Comment (jaybarra) [inline app/controllers/granules_controller.rb:61]: minor typo _seach_ instead of _search_

File: app/helpers/granules_helper.rb [hunk 2]
```diff
@@ -1,32 +1,309 @@
...
+    spatial_list = []
+
+    return spatial_list if metadata.nil?
+
+    horizontal_spatial_domain = metadata.fetch('horizontalSpatialDomain', {})
+
+    if horizontal_spatial_domain
+      geometry = horizontal_spatial_domain.fetch('geometry', {})
+
+      # orbit = horizontal_spatial_domain['orbit']
```
Comment (jaybarra) [inline app/helpers/granules_helper.rb:118]: Could these commented out sections be removed?

File: config/environments/production.rb [hunk 3]
```diff
@@ -67,19 +67,20 @@
   config.i18n.fallbacks = true
 
   # Send deprecation notices to registered listeners.
-  #config.active_support.deprecation = :notify
+  # config.active_support.deprecation = :notify
```
Comment (abbottry) [inline config/environments/production.rb:70]: why are these commented?

File: app/helpers/granules_helper.rb [hunk 4]
```diff
@@ -1,32 +1,309 @@
...
+  # @param [Hash] concept_id The concept id of the collection being requested
+  #
+  # @return [RestClient::Response] Response from GraphQL
+  def query_graphql(params)
+    url = Rails.configuration.graphql_endpoint
+
+    RestClient::Request.execute(
+      method: :post,
+      url: url,
+      headers: { accept: :json, content_type: :json, 'Client-Id': "opensearch-#{Rails.env.downcase}-client" },
```
Comment (abbottry) [inline app/helpers/granules_helper.rb:34]: Add `X-Request-Id`

File: cwic-holdings.json [hunk 5]
```diff
@@ -0,0 +1,984 @@
+  {
```
Comment (abbottry) [inline cwic-holdings.json:1]: delete

File: spec/controllers/granules_controller_spec.rb [hunk 6]
```diff
@@ -31,64 +33,74 @@
...
-      get :descriptor_document, :format => :xml, :params => { :clientId => '', :collectionConceptId => 'C1214603059-SCIOPS' }
-      expect(response.status).to eq(200)
-      expect(response).to render_template("granules/ccmeo.xml.erb")
-    end
-    it "renders an eumetsat OSDD" do
-      get :descriptor_document, :format => :xml, :params => { :clientId => '', :collectionConceptId => 'C1588876552-EUMETSAT' }
-      expect(response.status).to eq(200)
-      expect(response).to render_template("granules/eumetsat.xml.erb")
-    end
+    # it "renders a ccmeo OSDD" do
```
Comment (abbottry) [inline spec/controllers/granules_controller_spec.rb:51]: add comment as to why these commented

File: config/environments/uat.rb [hunk 7]
```diff
@@ -49,29 +49,29 @@
   config.log_level = :debug
 
   config.relative_url_root = '/opensearch'
+  config.graphql_endpoint = ENV['GRAPHQL_ENDPOINT']
+  config.cache_store = :redis_cache_store, { url: "redis://#{ENV['REDIS_URL']}:#{ENV['REDIS_PORT']}" }
 
-  config.cache_store = :memory_store, { size: 64.megabytes }
-  
   config.eosdis_providers = %w[
```
[the remark this answers is not in the document]
Comment (abbottry) [reply config/environments/uat.rb:55]: no

**Pull request merged**